spi/src/main/java/org/opensearch/jobscheduler/spi/schedule/CronSchedule.java

```diff
@@ -61,17 +61,26 @@
     private String expression;
     private ExecutionTime executionTime;
     private Clock clock;
+    private long scheduleDelay;
 
     public CronSchedule(String expression, ZoneId timezone) {
         this.expression = expression;
         this.timezone = timezone;
         this.executionTime = ExecutionTime.forCron(cronParser.parse(this.expression));
         clock = Clock.system(timezone);
+        this.scheduleDelay = 0;
+    }
+
+    public CronSchedule(String expression, ZoneId timezone, long scheduleDelay) {
+        this(expression, timezone);
+        this.scheduleDelay = scheduleDelay;
```

Comment:
minor/nitpick:

Can you move the setting logic to all fields to new method `CronSchedule(String expression, ZoneId timezone, long scheduleDelay)` and call this method with scheduleDelay as zero from `CronSchedule(String expression, ZoneId timezone)` that way we are not setting the schedule delay twice when method at line 74 is called.